### DISCUSSION TITLE
variables in variable statements?

### DISCUSSION THREAD

**Discussion by svenXY** [mistweaverco/kulala.nvim#270] [Q&A] [answered]
Would it make sense to make it possible to use variables within variable statements like the following, thus making it possible to put them into env files or imilar which would make it even more flexible?

```http
@user = "some_username"

# @env-stdin-cmd AID jq -r '.results[] | select(.desc == "{{user}}") | .id'
GET https://cloud.mongodb.com/api/atlas/v2/groups/{{project}}/apiKeys
```

**Comment by gorillamoe** [comment]
For these kind of computations, using scripts might be the perfect fit.

Can you try if this fits your needs?

https://kulala.mwco.app/docs/scripts/overview

**Reply by svenXY** [reply]
phew, that's a bit too fat for my tiny needs, but ok, there is a solution in case I need it, so thanks!

I was under the maybe oversimplified impression that it cannot be too hard to check the string in @env-stdin-cmd for variables before calling it

**Reply by gorillamoe** [reply]
Not tested, but this should do it, I don't consider this too overcomplicated, do you?

```http
@ENV_USER = "some_username"

GET https://cloud.mongodb.com/api/atlas/v2/groups/{{project}}/apiKeys
> {%
client.global.set("AID", data.results.filter(item => item.desc === request.environment.get("ENV_USER")).id);
%}
```

But I have to check if request is available post-request, otherwise I might just fix that when I'm back home.

**Comment by gorillamoe** [comment, marked as answer]
Implemented in v4.2.0:

```http
@ENV_USER = some_username
@ENV_PROJECT = project_name

POST https://httpbin.org/post HTTP/1.1
Content-Type: application/json
Accept: application/json

{
  "project": "{{ENV_PROJECT}}",
  "results": [
    {
      "id": 1,
      "desc": "some_username"
    },
    {
      "id": 2,
      "desc": "another_username"
    }
  ]
}

> {%
client.global.set("AID", response.body.json.results.filter(item => item.desc === request.environment.get("ENV_USER"))[0].id);
%}

###

POST https://httpbin.org/post HTTP/1.1
Content-Type: application/json
Accept: application/json

{
  "id": "{{AID}}"
}
```

**Reply by svenXY** [reply]
thanks. This looks much cleaner to me. To redirect into a variable seems much more intuitive to me than using @env-stdin-cmd above the request.

I'll see when I can try this out.

**Reply by gorillamoe** [reply]
Yes, I agree, but the `@env-stdin-cmd` is legacy code from I think v1, so maybe some people still use it :)
